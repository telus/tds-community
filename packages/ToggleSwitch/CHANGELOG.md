# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [3.1.0](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.5...@tds/community-toggle-switch@3.1.0) (2021-09-21)


### Features

* **community-toggle-switch:** react 17 and styled components 5 ([2874526](https://github.com/telus/tds-community/commit/2874526377eac43c6045d4aabcef91df67aaa8fa))





## [3.0.5](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.4...@tds/community-toggle-switch@3.0.5) (2020-09-09)


### Bug Fixes

* **community-toggle-switch:** update util-helpers version ([f0ee4cb](https://github.com/telus/tds-community/commit/f0ee4cb34205373061f3d19839d8a2929e70d351))





## [3.0.4](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.3...@tds/community-toggle-switch@3.0.4) (2020-07-31)

**Note:** Version bump only for package @tds/community-toggle-switch





## [3.0.3](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.2...@tds/community-toggle-switch@3.0.3) (2020-04-13)

**Note:** Version bump only for package @tds/community-toggle-switch





## [3.0.2](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.1...@tds/community-toggle-switch@3.0.2) (2020-03-19)

**Note:** Version bump only for package @tds/community-toggle-switch





## [3.0.1](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@3.0.0...@tds/community-toggle-switch@3.0.1) (2020-01-31)


### Bug Fixes

* **community-toggle-switch:** makes spinnerLabel prop optional ([ee04d07](https://github.com/telus/tds-community/commit/ee04d07))





# [3.0.0](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@2.0.0...@tds/community-toggle-switch@3.0.0) (2019-11-22)


### Features

* **community-toggle-switch:** idd sLoading and ref props ([affec8e](https://github.com/telus/tds-community/commit/affec8e))


### BREAKING CHANGES

* **community-toggle-switch:** Toggle no longer automatically sets focus to itself. Manually control focus using ref.





# [2.0.0](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@1.0.4...@tds/community-toggle-switch@2.0.0) (2019-09-30)


### Bug Fixes

* **community-toggle-switch:** design changes and refactoring ([f813eeb](https://github.com/telus/tds-community/commit/f813eeb))
* **community-toggle-switch:** making adjustments to padding ([bb20e07](https://github.com/telus/tds-community/commit/bb20e07))


### Features

* **community-toggle-switch:** new design changes for toggleSwitch ([be03ce3](https://github.com/telus/tds-community/commit/be03ce3)), closes [#112](https://github.com/telus/tds-community/issues/112)


### BREAKING CHANGES

* **community-toggle-switch:** add styled-components to peerDependencies
* **community-toggle-switch:** default style changes
* **community-toggle-switch:** New design layout for toggleSwitch, styled-components is now a peer dependency





## [1.0.4](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@1.0.3...@tds/community-toggle-switch@1.0.4) (2019-09-17)

**Note:** Version bump only for package @tds/community-toggle-switch





## [1.0.3](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@1.0.2...@tds/community-toggle-switch@1.0.3) (2019-06-13)

**Note:** Version bump only for package @tds/community-toggle-switch





## [1.0.2](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@1.0.1...@tds/community-toggle-switch@1.0.2) (2019-01-08)

**Note:** Version bump only for package @tds/community-toggle-switch

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telus/tds-community/compare/@tds/community-toggle-switch@1.0.0...@tds/community-toggle-switch@1.0.1) (2018-10-22)

**Note:** Version bump only for package @tds/community-toggle-switch

<a name="1.0.0"></a>

# 1.0.0 (2018-10-10)

### Bug Fixes

- **community-toggle-switch:** added missing jest snapshot ([39d3564](https://github.com/telus/tds-community/commit/39d3564))
- **community-toggle-switch:** lerna add dependencies from [@tds](https://github.com/tds)/core-\* ([4013772](https://github.com/telus/tds-community/commit/4013772))

### Features

- **community-toggle-switch:** initial commit for component ([198cdbf](https://github.com/telus/tds-community/commit/198cdbf))
