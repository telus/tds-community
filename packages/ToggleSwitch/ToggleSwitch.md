```jsx
<ToggleSwitch id="42" name="name" value="value" label="Enable data" />
```
