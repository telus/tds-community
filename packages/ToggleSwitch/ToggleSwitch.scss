@import '~@tds/core-colours/colours';

.switch {
  position: relative;
  cursor: pointer;
  display: inline-block;
  width: 44px;
  min-width: 44px;
  height: 22px;
  border-radius: 22px;
  background-color: $color-gainsboro;
}

.slider {
  position: absolute;
  width: 18px;
  height: 18px;
  top: 2px;
  left: 2px;
  border-radius: 18px;
  background-color: white;
  transition: left 0.25s;
}

.switchOn {
  background-color: $color-accessible-green;
  .slider {
    left: 24px;
  }
}

.switchOff {
  background-color: $color-gainsboro;
  .slider {
    left: 2px;
  }
}

.switchDisabled {
  background-color: $color-gainsboro;
}
.switchDisabled * {
  cursor: not-allowed !important;
}

.hiddenInput {
  opacity: 0;
}
.hiddenInput:focus + .switch {
  box-shadow: 0 0 4px 1px $color-shuttle-grey;
}
