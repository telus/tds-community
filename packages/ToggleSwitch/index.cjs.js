require('./dist/index.css')
const ToggleSwitch = require('./dist/index.cjs')

module.exports = ToggleSwitch
