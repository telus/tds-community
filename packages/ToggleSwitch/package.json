{
  "name": "@tds/community-toggle-switch",
  "version": "3.0.1",
  "description": "",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telus/tds-community.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telus/tds-community/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/core-box": "^2.1.2",
    "@tds/core-colours": "^2.2.1",
    "@tds/core-spinner": "^3.1.11",
    "@tds/core-text": "^3.0.4",
    "@tds/core-tooltip": "^4.0.8",
    "@tds/util-helpers": "^1.2.0",
    "prop-types": "^15.6.2"
  }
}
