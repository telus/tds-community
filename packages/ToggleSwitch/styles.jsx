import styled from 'styled-components'
import { colorAccessibleGreen, colorGainsboro } from '@tds/core-colours'

export const StyledLabel = styled.label({
  display: 'flex',
  justifyContent: 'space-between',
})

StyledLabel.displayName = 'StyledLabel'

/**
 * Button background should only turn green when the `checked` prop is `true`
 */

export const Button = styled.button(props => ({
  appearance: 'none',
  background: 'none',
  boxShadow: 'none',
  position: 'relative',
  border: 'none',
  padding: '0',
  width: '40px',
  height: '24px',
  borderRadius: '24px',
  backgroundColor: props['aria-checked'] ? colorAccessibleGreen : colorGainsboro,
  '&:focus': {
    outlineStyle: props.isLoading ? 'none' : 'solid',
    outlineColor: 'rgb(59, 153, 252)',
    outlineWidth: '5px',
  },
  '&:focus::-moz-focus-inner': {
    border: 0,
  },
}))

export const Slider = styled.span(props => ({
  position: 'absolute',
  width: '18px',
  height: '18px',
  top: '3px',
  left: props.pressed ? '20px' : '3px',
  borderRadius: '18px',
  backgroundColor: 'white',
  transition: 'left 0.25s',
  boxShadow: '0 0 2px 0 #000',
}))

export const InputSwitchWrapper = styled.div({
  display: 'flex',
  justifyContent: 'center',
})
