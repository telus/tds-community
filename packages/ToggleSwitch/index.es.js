import './dist/index.css'
import ToggleSwitch from './dist/index.es'

export default ToggleSwitch
