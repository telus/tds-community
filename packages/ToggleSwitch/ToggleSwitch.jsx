import React, { useState } from 'react'
import PropTypes from 'prop-types'

import { safeRest } from '@tds/util-helpers'
import Text from '@tds/core-text'
import Box from '@tds/core-box'
import Tooltip from '@tds/core-tooltip'
import Spinner from '@tds/core-spinner'

import { StyledLabel, Button, Slider, InputSwitchWrapper } from './styles'
import warn from '../../shared/utils/warn'

/**
 * @version ./package.json
 */

const ToggleSwitch = ({
  id,
  label,
  tooltipText,
  checked,
  onClick,
  tooltipCopy,
  spinnerLabel,
  ...rest
}) => {
  if (tooltipText && !tooltipCopy) {
    warn('@tds/community-toggle-switch', 'You must provide tooltipCopy when using tooltipText')
  }

  const labelledById = `${id}-label`

  const [isPressed, setIsPressed] = useState(checked)
  const [isLoading, setIsLoading] = useState(false)
  const [isSpinning, setIsSpinning] = useState(false)

  React.useEffect(() => {
    setIsSpinning(false)
    setIsLoading(false)
  }, [checked])

  React.useEffect(() => {
    const timer = setTimeout(() => {
      if (isLoading) {
        setIsSpinning(true)
      } else {
        setIsSpinning(false)
      }
    }, 250) // time needed for a slider to move
    return () => clearTimeout(timer)
  }, [isLoading])

  const handleClick = event => {
    setIsPressed(!isPressed)
    setIsLoading(true)
    onClick(event)
  }

  return (
    <StyledLabel htmlFor={id}>
      <Box inline between={2}>
        <Text id={labelledById} size="medium">
          {label}
        </Text>
        {tooltipText && tooltipCopy && <Tooltip copy={tooltipCopy}>{tooltipText}</Tooltip>}
      </Box>
      <InputSwitchWrapper>
        <Spinner tag="span" spinning={isSpinning} label={spinnerLabel} size="small" inline>
          <Button
            {...safeRest(rest)}
            id={id}
            role="switch"
            aria-checked={checked}
            aria-labelledby={labelledById}
            data-testid={`${id}-switch`}
            onClick={!isLoading ? handleClick : null}
          >
            <Slider pressed={isPressed} />
          </Button>
        </Spinner>
      </InputSwitchWrapper>
    </StyledLabel>
  )
}

ToggleSwitch.propTypes = {
  /** The unique HTML id for this form input. */
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,

  /** The label. */
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,

  /** The aria-checked state of a button. */
  checked: PropTypes.bool,

  /** Text written for TDS Tooltip. */
  tooltipText: PropTypes.string,

  /** Language provided to the copy prop in TDS Tooltip (en, fr). */
  tooltipCopy: PropTypes.string,

  /** Communicates a message to assistive technology while spinner is visible. */
  spinnerLabel: PropTypes.string.isRequired,

  /** A callback function to be invoked when the ToggleSwitch button is clicked on.
   @param {SyntheticEvent} event The React `SyntheticEvent` */
  onClick: PropTypes.func.isRequired,
}

ToggleSwitch.defaultProps = {
  checked: false,
  tooltipText: undefined,
  tooltipCopy: undefined,
}

export default ToggleSwitch
