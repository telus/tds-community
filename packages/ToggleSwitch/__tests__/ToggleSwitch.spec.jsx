import React from 'react'
import { shallow } from 'enzyme'

import ToggleSwitch from '../ToggleSwitch'

describe('ToggleSwitch', () => {
  const defaultProps = { id: 'my-id-123', label: 'label', name: 'name', value: 'value' }

  const doShallow = props => shallow(<ToggleSwitch {...defaultProps} {...props} />)

  it('renders', () => {
    const toggleSwitch = doShallow()

    expect(toggleSwitch).toMatchSnapshot()
  })

  it('should have for attribute matching passed id', () => {
    const toggleSwitch = doShallow()

    expect(
      toggleSwitch
        .find('label')
        .at(0)
        .prop('htmlFor')
    ).toEqual(defaultProps.id)
  })

  it('should have label text with an id that matches the checkbox aria-labelledby attribute', () => {
    const toggleSwitch = doShallow()

    expect(
      toggleSwitch
        .find(`#${defaultProps.id}`)
        .at(0)
        .prop('aria-labelledby')
    ).toEqual(`${defaultProps.id}-label`)

    expect(
      toggleSwitch
        .find(`#${defaultProps.id}`)
        .at(0)
        .prop('aria-labelledby')
    ).toEqual(
      toggleSwitch
        .find(`#${defaultProps.id}-label`)
        .at(0)
        .prop('id')
    )
  })

  it('should be default unchecked', () => {
    const toggleSwitch = doShallow()

    expect(
      toggleSwitch
        .find('.hiddenInput')
        .at(0)
        .prop('checked')
    ).toEqual(false)

    expect(
      toggleSwitch
        .find(`[data-testid="${defaultProps.id}-switch"]`)
        .at(0)
        .prop('aria-checked')
    ).toEqual(false)
  })

  it('should be checked when checked property is true', () => {
    const toggleSwitch = doShallow({ checked: true })
    expect(
      toggleSwitch
        .find(`#${defaultProps.id}`)
        .at(0)
        .prop('checked')
    ).toEqual(true)

    expect(
      toggleSwitch
        .find(`[data-testid="${defaultProps.id}-switch"]`)
        .at(0)
        .prop('aria-checked')
    ).toEqual(true)
  })

  it('internal state should toggle checked when the input is changed by default', () => {
    const toggleSwitch = doShallow()

    toggleSwitch
      .find(`#${defaultProps.id}`)
      .at(0)
      .simulate('change')

    expect(
      toggleSwitch
        .find(`#${defaultProps.id}`)
        .at(0)
        .prop('checked')
    ).toEqual(true)

    expect(
      toggleSwitch
        .find(`[data-testid="${defaultProps.id}-switch"]`)
        .at(0)
        .prop('aria-checked')
    ).toEqual(true)
  })

  it('should override internal state change when onChange handler is provided ', () => {
    const mockOnChange = jest.fn()
    const mockEvent = { persist: () => {} }
    const toggleSwitch = doShallow({ checked: false, onChange: mockOnChange })

    toggleSwitch
      .find(`#${defaultProps.id}`)
      .at(0)
      .simulate('change', mockEvent)

    // Checked state should remain false instead of toggling
    expect(
      toggleSwitch
        .find(`#${defaultProps.id}`)
        .at(0)
        .prop('checked')
    ).toEqual(false)

    expect(
      toggleSwitch
        .find(`[data-testid="${defaultProps.id}-switch"]`)
        .at(0)
        .prop('aria-checked')
    ).toEqual(false)

    // Custom onChange handler should get event delgated to it instead
    expect(mockOnChange).toHaveBeenCalledWith(mockEvent)
  })

  it('should call onFocus handler when provided ', () => {
    const mockOnFocus = jest.fn()
    const mockEvent = { persist: () => {} }
    const toggleSwitch = doShallow({ checked: false, onFocus: mockOnFocus })

    toggleSwitch
      .find(`#${defaultProps.id}`)
      .at(0)
      .simulate('focus', mockEvent)

    expect(mockOnFocus).toHaveBeenCalledWith(mockEvent)
  })

  it('should call onBlur handler when provided ', () => {
    const mockOnBlur = jest.fn()
    const mockEvent = { persist: () => {} }
    const toggleSwitch = doShallow({ checked: false, onBlur: mockOnBlur })

    toggleSwitch
      .find(`#${defaultProps.id}`)
      .at(0)
      .simulate('blur', mockEvent)

    expect(mockOnBlur).toHaveBeenCalledWith(mockEvent)
  })

  it('passes additional attributes to the element', () => {
    const toggleSwitch = doShallow({ 'data-some-attr': 'some value' })

    expect(toggleSwitch.find(`#${defaultProps.id}`).at(0)).toHaveProp(
      'data-some-attr',
      'some value'
    )
  })
})
