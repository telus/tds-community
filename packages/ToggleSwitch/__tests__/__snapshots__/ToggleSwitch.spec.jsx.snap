// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  box-shadow: none;
  position: relative;
  border: none;
  padding: 0;
  width: 2.5rem;
  height: 1.5rem;
  border-radius: 1.5rem;
  background-color: #d8d8d8;
}

.c2:focus {
  outline-style: solid;
  outline-color: rgb(59,153,252);
  outline-width: 0.3125rem;
}

.c2:focus::-moz-focus-inner {
  border: 0;
}

.c3 {
  position: absolute;
  width: 1.125rem;
  height: 1.125rem;
  top: 0.1875rem;
  left: 0.1875rem;
  border-radius: 1.125rem;
  background-color: white;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  box-shadow: 0 0 2px 0 #000;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (prefers-reduced-motion:reduce) {
  .c3 {
    -webkit-transition: none !important;
    transition: none !important;
  }
}

<ToggleSwitch
  checked={false}
  id="my-id-123"
  isLoading={false}
  label="label"
  name="name"
  onClick={[MockFunction]}
  toolTipCopy="en"
  value="value"
>
  <StyledLabel
    htmlFor="my-id-123"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__StyledLabel-i9h5g3-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "display: flex; justify-content: space-between;",
            ],
          },
          "displayName": "StyledLabel",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__StyledLabel-i9h5g3-0",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      htmlFor="my-id-123"
    >
      <label
        className="c0"
        htmlFor="my-id-123"
      >
        <Box
          between={2}
          inline={true}
          tag="div"
        >
          <div
            className="TDS_Box-modules__betweenRightMargin-2___20pzC TDS_Box-modules__inline___jTHcz"
          >
            <Text
              block={false}
              bold={false}
              id="my-id-123-label"
              invert={false}
              size="medium"
            >
              <span
                className="TDS_Typography-modules__default___KlcpP TDS_Typography-modules__medium___1rxfE TDS_Typography-modules__wordBreak___3OZx_ TDS_Typography-modules__mediumFont___XMrRj TDS_Typography-modules__color___2CNH8"
                id="my-id-123-label"
              >
                label
              </span>
            </Text>
          </div>
        </Box>
        <styles__InputSwitchWrapper>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__InputSwitchWrapper-i9h5g3-3",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "display: flex; justify-content: center;",
                  ],
                },
                "displayName": "styles__InputSwitchWrapper",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__InputSwitchWrapper-i9h5g3-3",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <Spinner
                a11yLabel="A spinner is active. Please wait while the page completes a task."
                dangerouslyHideVisibleLabel={false}
                fullScreen={false}
                inline={true}
                size="small"
                spinning={false}
                tag="span"
                variant="primary"
              >
                <styles__Button
                  aria-checked={false}
                  aria-labelledby="my-id-123-label"
                  data-testid="my-id-123-switch"
                  id="my-id-123"
                  name="name"
                  onClick={[Function]}
                  role="switch"
                  toolTipCopy="en"
                  value="value"
                >
                  <StyledComponent
                    aria-checked={false}
                    aria-labelledby="my-id-123-label"
                    data-testid="my-id-123-switch"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "styles__Button-i9h5g3-1",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            [Function],
                          ],
                        },
                        "displayName": "styles__Button",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "styles__Button-i9h5g3-1",
                        "target": "button",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={
                      Object {
                        "current": .c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  box-shadow: none;
  position: relative;
  border: none;
  padding: 0;
  width: 2.5rem;
  height: 1.5rem;
  border-radius: 1.5rem;
  background-color: #d8d8d8;
}

.c0:focus {
  outline-style: solid;
  outline-color: rgb(59,153,252);
  outline-width: 0.3125rem;
}

.c0:focus::-moz-focus-inner {
  border: 0;
}

.c1 {
  position: absolute;
  width: 1.125rem;
  height: 1.125rem;
  top: 0.1875rem;
  left: 0.1875rem;
  border-radius: 1.125rem;
  background-color: white;
  -webkit-transition: left 0.25s;
  transition: left 0.25s;
  box-shadow: 0 0 2px 0 #000;
}

@media (prefers-reduced-motion:reduce) {
  .c1 {
    -webkit-transition: none !important;
    transition: none !important;
  }
}

<button
                          aria-checked="false"
                          aria-labelledby="my-id-123-label"
                          class="c0"
                          data-testid="my-id-123-switch"
                          id="my-id-123"
                          name="name"
                          role="switch"
                          value="value"
                        >
                          <span
                            class="c1"
                          />
                        </button>,
                      }
                    }
                    id="my-id-123"
                    name="name"
                    onClick={[Function]}
                    role="switch"
                    toolTipCopy="en"
                    value="value"
                  >
                    <button
                      aria-checked={false}
                      aria-labelledby="my-id-123-label"
                      className="c2"
                      data-testid="my-id-123-switch"
                      id="my-id-123"
                      name="name"
                      onClick={[Function]}
                      role="switch"
                      value="value"
                    >
                      <styles__Slider
                        pressed={false}
                      >
                        <StyledComponent
                          forwardedComponent={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "attrs": Array [],
                              "componentStyle": ComponentStyle {
                                "componentId": "styles__Slider-i9h5g3-2",
                                "isStatic": false,
                                "lastClassName": "c3",
                                "rules": Array [
                                  [Function],
                                ],
                              },
                              "displayName": "styles__Slider",
                              "foldedComponentIds": Array [],
                              "render": [Function],
                              "styledComponentId": "styles__Slider-i9h5g3-2",
                              "target": "span",
                              "toString": [Function],
                              "warnTooManyClasses": [Function],
                              "withComponent": [Function],
                            }
                          }
                          forwardedRef={null}
                          pressed={false}
                        >
                          <span
                            className="c3"
                          />
                        </StyledComponent>
                      </styles__Slider>
                    </button>
                  </StyledComponent>
                </styles__Button>
              </Spinner>
            </div>
          </StyledComponent>
        </styles__InputSwitchWrapper>
      </label>
    </StyledComponent>
  </StyledLabel>
</ToggleSwitch>
`;
